nomad/structs/config/consul.go

```diff
@@ -359,6 +401,74 @@ func (c *ConsulConfig) Copy() *ConsulConfig {
 	if nc.AllowUnauthenticated != nil {
 		nc.AllowUnauthenticated = pointer.Of(*nc.AllowUnauthenticated)
 	}
+	if nc.UseIdentity != nil {
+		nc.UseIdentity = pointer.Of(*nc.UseIdentity)
+	}
 
 	return nc
 }
+
+// WorkloadIdentity is the jobspec block which determines if and how a workload
+// identity is exposed to tasks similar to the Vault block.
+//
+// This is a copy of WorkloadIdentity from nomad/structs package in order to
+// avoid import cycles.
+type WorkloadIdentity struct {
+	Name string
+
+	// Audience is the valid recipients for this identity (the "aud" JWT claim)
+	// and defaults to the identity's name.
+	Audience []string
+
+	// Env injects the Workload Identity into the Task's environment if
+	// set.
+	Env bool
+
+	// File writes the Workload Identity into the Task's secrets directory
+	// if set.
+	File bool
+
+	// ServiceName is used to bind the identity to a correct Consul service.
+	ServiceName string
```

Comment:
This field now makes me wonder that this is not quite the same as the `WorkloadIdentity` in `structs`. I think it would never make sense to set this as an agent config? 🤔 

We could leave it in to keep the structs in sync, but maybe we need to make sure it's not set? Otherwise all service identities would have the same `ServiceName`

Reply:
I made `mapstructure` and `json` decoders ignore the field for agent config.